### ISSUE TITLE
All projects: Configure top level maven surefire configuration to include all tests

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#8881]
Some tests are simply not being run on CI.

The default behaviour of Surefire is supposed to be as follows:
```
By default, the Surefire Plugin will automatically include all test classes with the following wildcard patterns:

"**/Test*.java" - includes all of its subdirectories and all Java filenames that start with "Test".
"**/*Test.java" - includes all of its subdirectories and all Java filenames that end with "Test".
"**/*Tests.java" - includes all of its subdirectories and all Java filenames that end with "Tests".
"**/*TestCase.java" - includes all of its subdirectories and all Java filenames that end with "TestCase".
```
https://maven.apache.org/surefire/maven-surefire-plugin/examples/inclusion-exclusion.html

Here's a snippet of test output: https://gist.github.com/AlexDBlack/c281eab3d13108f75963316d3c9a3bdf
Taken from: https://jenkins.konduit.ai/blue/organizations/jenkins/konduitai%2Fdeeplearning4j/detail/PR-412/3/pipeline/648

Note for nd4j-parameter-server-client: 2 tests run only:
```
[2020-04-25T10:57:44.985Z] -------------------------------------------------------
[2020-04-25T10:57:44.985Z]  T E S T S
[2020-04-25T10:57:44.985Z] -------------------------------------------------------
[2020-04-25T10:57:44.985Z] Running org.nd4j.parameterserver.client.ParameterServerClientTest
[2020-04-25T10:57:44.985Z] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.032 sec
[2020-04-25T10:57:44.985Z] Running org.nd4j.parameterserver.client.ParameterServerClientPartialTest
[2020-04-25T10:57:44.985Z] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 sec
[2020-04-25T10:57:44.985Z] 
[2020-04-25T10:57:44.985Z] Results :
[2020-04-25T10:57:44.985Z] 
[2020-04-25T10:57:44.985Z] Tests run: 2, Failures: 0, Errors: 0, Skipped: 2
```

Yet the following test (been on master for a long time) is NOT run:
https://github.com/eclipse/deeplearning4j/blob/master/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-client/src/test/java/org/nd4j/parameterserver/background/RemoteParameterServerClientTests.java

It should be included as it matches the pattern `**/*Tests.java` - but it clearly isn't run.

Adding maven surefire configuration as per here makes it be included:
https://github.com/eclipse/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-tests/pom.xml#L169-L187
(Note that in practice we probably only need the first 2 patterns)

```
[2020-04-25T11:15:58.894Z] Running org.nd4j.parameterserver.background.RemoteParameterServerClientTests
[2020-04-25T11:15:58.894Z] o.n.c.t.BaseND4JTest - RemoteParameterServerClientTests.remoteTests
```

Unsurprisingly, it fails because it's probably never actually run on CI before :/

tl;dr: We really need to have the maven surefire "include everything" configuration at the top-level pom (if possible) so it's inherited by all projects to avoid this happening again (and to discover any other tests that haven't been running on CI).

**Comment by AlexDBlack** [comment]
Another instance of this: datavec-data-image
According to Maven: 47 tests
```
Tests run: 47, Failures: 0, Errors: 0, Skipped: 1
```
According to Intellij: 56 tests
![image](https://user-images.githubusercontent.com/2360237/80935507-47983600-8e10-11ea-93d6-46a896aa03be.png)
